src/components/KInput/KInput.vue

```diff
@@ -324,13 +324,16 @@ export default {
   box-shadow: none !important;
 
   &.has-icon {
+    $kInputLineHeight: var(--kui-line-height-40, $kui-line-height-40);
+    // TODO: this block is repetitive and can be refactored into a mixin
     // input size medium
-    $kInputMediumSizingX: 10px;
-    $kInputMediumSizingY: var(--spacing-md, spacing(md));
-    $kInputMediumIconSize: 24px;
+    $kInputMediumSizingY: var(--kui-space-40, $kui-space-40);
+    $kInputMediumSizingX: var(--spacing-md, var(--kui-space-60, $kui-space-60));
+    $kInputMediumIconSize: var(--kui-icon-size-50, $kui-icon-size-50);
 
     ~ .input-icon {
-      top: $kInputMediumSizingX;
+      // (height of entire element (paddings + line height) - icon size) / 2
+      top: calc((($kInputMediumSizingY + $kInputMediumSizingY + $kInputLineHeight) - $kInputMediumIconSize) / 2);
```

Comment:
question: this logic is very different than what was here before...?

Same comment applies to all the other blocks